Sources/SWBCore/SpecImplementations/Tools/CCompiler.swift

```diff
@@ -1648,6 +1651,22 @@ public class ClangCompilerSpec : CompilerSpec, SpecIdentifierType, GCCCompatible
         }
     }
 
+    func addCompilerMetadataFlags(_ cbc: CommandBuildContext, _ outputPath: Path, _ delegate: any TaskGenerationDelegate, _ taskOutputs: inout [any PlannedNode]) -> [String] {
+        guard cbc.scope.evaluate(BuiltinMacros.EMIT_COMPILER_SOURCE_METADATA) else  {
+            return []
+        }
+
+        guard let metadatatype = cbc.producer.lookupFileType(identifier: "text.json.compiler-metadata.source") else {
+            return []
+        }
+
+        let securityMetadataNode = delegate.createNode(outputPath)
+        let ftb = FileToBuild(absolutePath: securityMetadataNode.path, fileType: metadatatype)
+        taskOutputs.append(securityMetadataNode)
+        delegate.declareOutput(ftb)
+        return ["-fdiagnostics-add-output=sarif:file=" + securityMetadataNode.path.str]
```

Comment:
Do you need to add quoting and/or escaping here, since the path has to be part of the same option? Or does the fact the result is an array mean that the build system will property quote each array element?